src/dimension.c

```diff
@@ -129,11 +134,11 @@ dimension_fill_in_from_tuple(Dimension *d, TupleInfo *ti, Oid main_table_relid)
 		   DatumGetName(values[Anum_dimension_column_name - 1]),
 		   NAMEDATALEN);
 
-	if (d->type == DIMENSION_TYPE_CLOSED)
+	if (!isnull[Anum_dimension_partitioning_func_schema - 1] &&
+		!isnull[Anum_dimension_partitioning_func - 1])
```

Comment:
Is it valid to have only one of these null? Should we have a warning in that case?

Reply:
Probably an Assert in that case because it is an error. Our schema actually protects against such situations, and I do believe we double check this when creating a dimension so we need not recheck everywhere.